[DOC] Updates from working session on Helm docs (#2590)
grafana/tempo: docs/sources/helm-charts/tempo-distributed/get-started-helm-charts/_index.md

   helm repo update
   ```

   {{% admonition type="note" %}} The Helm chart at https://grafana.github.io/helm-charts is a publication of the source code at grafana/tempo.
   {{% /admonition %}}

## Set Helm chart values

> Your Helm chart values are set in the `custom.yaml` file.

Reviewer: ```suggestion
Your Helm chart values are set in the file `custom.yaml`, which you will create in a working directory in the set of steps below.
```

> The Tempo Helm char values example `custom.yaml` file sets the storage and traces options, enables the gateway, and sets the cluster to main.

Reviewer: ```suggestion
The `custom.yaml` file sets the storage and traces options, enables the gateway, and sets the cluster to main.
```

The `traces` configure the distributor's receiver protocols.

To customize your Helm chart values:

> 1. Create a `custom.yaml` file.

Reviewer: ```suggestion
1. Create a `custom.yaml` file in your working directory.
```

> 1. Copy and paste either the Tempo Helm chart values or the GET Helm chart values into your file.

Reviewer: ```suggestion
1. From the examples below, copy and paste either the Tempo Helm chart values or the Grafana Enterprise Traces Helm chart values into your file.
```

> 1. Save the file in your working directory.

Reviewer: ```suggestion
1. Save your `custom.yaml` file in your working directory.
```

> 1. Set your storage values, the example helm chart values points to the MinIO instance configured by the chart.

Reviewer: ```suggestion
1. For simple deployments, leave the MinIO settings as they are; MinIO will be deployed for you by the Helm chart. If instead you would like to use a different object store, adjust the storage values as appropriate for your environment; [see this page](https://grafana.com/docs/tempo/latest/configuration/#storage) for reference and examples.
```

> 1. Set your traces values to configure the receivers on the Tempo distributor.

Reviewer: I'm not sure what to suggest here but I feel we should add some clarification about whether they can just flip the true/false values as needed for their environment, and any other needed clarification, and/or a pointer to a docs page that might add clarification at this point.

> 1. Set your traces values to configure the receivers on the Tempo distributor.

Reviewer: ```suggestion
1. Set your traces values to configure the receivers on the Tempo distributor. (More detail on this is further down on this page.)
```

> 1. Create a `custom.yaml` file.
> 1. Copy and paste either the Tempo Helm chart values or the GET Helm chart values into your file.
> 1. Save the file in your working directory.
> 1. Set your storage values, the example helm chart values points to the MinIO instance configured by the chart.
> 1. Set your traces values to configure the receivers on the Tempo distributor.
> 1. Save the changes to your file.

Reviewer: ```suggestion
1. Create a `custom.yaml` file.
1. Copy and paste either the Tempo Helm chart values or the GET Helm chart values into your file.
1. Save the file in your working directory.
1. Set your storage values, the special example helm chart values points to the MinIO instance configured by the chart.
1. Set your traces values to configure the receivers on the Tempo distributor.
1. Save the changes to your file.
```

### Tempo Helm chart values

This sample file contains example values for installing Tempo using Helm.

> 

Author: @zachfi  Question about helm configuration for MinIO. Tempo Helm chart has values that aren't configured for GET Helm chart. Do we need to copy the MinIO section from the Tempo chart values sample to the GET Helm chart value sample? 

The Tempo Helm chart has these values for MinIO (lines 116-130): 
```
minio:
  enabled: true
  mode: standalone
  rootUser: grafana-tempo
  rootPassword: supersecret
  buckets:
    # Default Tempo storage bucket.
    - name: tempo-traces
      policy: none
      purge: false
```

GET Helm chart values (lines 160-1): 
```
minio:
  enabled: true
```

Reviewer 2: Can you link where you see?

Author: GET values: https://github.com/grafana/tempo/pull/2590/files#r1244287106
Tempo values: https://github.com/grafana/tempo/pull/2590/files#r1244288148

Reviewer 2: Oh sorry, I see now you mean in the docs not in the chart.  Yes the same logic applies.  If users want to use minio, with non-default credentials, they will need to set them in the `minio` block as well as configured for the `s3` storage.

Author: I'll add this to the PR.

Reviewer 2: Okay cool.  Let me know when you're ready for a review.

Author: @zalegrala updated. Review?

Reviewer 3: Hello, sorry to barge into the convo like that. I hope it is okay for me to ask a short question regarding the docs here...

I ran `helm install` with the suggested custom values from the doc. However, multiple pods like the tempo-ingester fail to start up with the logs:

> level=info ts=2023-07-13T20:05:32.435682845Z caller=main.go:215 msg="initialising OpenTracing tracer"
> level=info ts=2023-07-13T20:05:32.450606561Z caller=main.go:102 msg="Starting Tempo" version="(version=2.1.1, branch=HEAD, revision=4157d7620)"
> level=error ts=2023-07-13T20:05:32.452641937Z caller=main.go:105 msg="error running Tempo" err="failed to init module services error initialising module: store: failed to create store unexpected error from ListObjects on tempo-traces: The Access Key Id you provided does not exist in our records."

Am I misunderstanding the docs? From what I see on my cluster, the actual access key and secret key which I can use to log in to MinIO are base64 encoded in the secret `tempo-minio` and are not equal to the strings `grafana-tempo` and `supersecret` specified in `storage.s3.access_key` and `storage.s3.secret_key`.

Reviewer 2: @ViviLearns2Code I think there was a mistake in the chart dependency version.  https://github.com/grafana/helm-charts/pull/2509 should correct the dependency version.  I'll test with @danstadler-pdx this afternoon.

Reviewer 2: @knylander-grafana For the two minio examples, one includes the root credentials and one does not.  If we don't include the root credentials as an example, folks might be confused how to change those credentials, but since the values there are default then we don't need to include them.  The only required key is the `minio.enabled: true` and then default values will be used.

Author: @zalegrala The example was updated to include the credentials (see [section below](https://github.com/grafana/tempo/pull/2590/files#r1265465821)

      bucket: 'tempo-traces'
      endpoint: 'tempo-minio:9000'
      insecure: true
#MinIO storage configuration
minio:
  enabled: true
  mode: standalone
  rootUser: grafana-tempo
  rootPassword: supersecret
  buckets:
    # Default Tempo storage bucket
    - name: tempo-traces
      policy: none

>       purge: false

Author: @zalegrala MinIO Tempo values

  enabled: true
gateway:
  enabled: false
# MinIO storage configuration
minio:

>   enabled: true

Author: @zalegrala This is the GET MinIO values.

Author: @zalegrala MinIO section updated with the values for the user and password, as well as buckets.

Reviewer 2: This looks good and matches below for the `storage.admin` section.
